src/api/report/reportRepository.ts

```diff
@@ -564,10 +564,10 @@ export const reportRepository = {
               timeToReachThreshold.full.push(diff);
             }
             threshold = 4;
+            allocationUsed = allocationUsed - BigInt(allocation);
```

Comment:
I think this doesn't do anything. When `allocationUsed >= allocation`, a.k.a. when we've processed enough deals to cover current allocation, we:
1. set the allocationUsed to `allocationUsed - BigInt(allocation);`
2. break the deals loop in next line, which means we go to the next iteration of allocations loop
3. and the first thing we do in is reset `allocationUsed` back to 0: https://github.com/fidlabs/allocator-compliance-checker/pull/23/files#diff-e2ebd8c3aded5c7bd2a5737a4c0df63444c24960c66b331eed57ae7f133599daR526

Instead of messing with allocationUsed, we should mess with the deal. Maybe substract the amount used for finishing current allocation from `clientDeals.deal_value` and push `dealIdx` back by one, so that for the new allocation we reprocess the same deal (but now with deal_value reduced)?